traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -1009,7 +1013,12 @@ func getMidServers(edgeIDs []int, servers map[int]tc.ServerNullable, tx *sqlx.Tx
 	WHERE s.id IN (?)))
 	`
 
-	query, args, err := sqlx.In(q, edgeIDs)
+	if cdnID > 0 {
+		q += ` AND s.cdn_id = ?`
+		filters = append(filters, cdnID)
+	}
```

Comment:
Checking if `cdnID` is greater than 0 should be unnecessary because servers are always in CDNs

Reply:
i think you're missing the point here. ONLY filter by cdn if cdn is passed in...

sometimes you don't want to filter by cdn...

Reply:
for example, if i say "give me all the mids associated with these 5 edges", it possible those 5 edges are in different cdns, so in that case, don't even worry about the cdn. however, it turns out that the only use of this method is in the context of a DS, where you'd always want to filter by cdn...

Reply:
this problem imo all stems back to the fact that servers from different cdns can live in the same cachegroup...

Reply:
> sometimes you don't want to filter by cdn...

Oh right, I forgot about that. I was in the `dsId` zone and was only thinking about that part.